### ISSUE TITLE
Sensitivity analysis of EE displacement parameters

### ISSUE THREAD

**Issue by gilliM** [swisstopo/topo-satromo#16]
Analise visuals and metrics related to changing parameters of the displacement function:

- referenceImage, 
- maxOffset, 
- projection, 
- patchWidth, 
- stiffness

**Comment by davidoesch** [comment]
Key, besides positional accuracy , is keeping the EECU time low. for a simple image as shown in https://radiantearth.github.io/stac-browser/#/external/d29cp2gnktw6by.cloudfront.net/stac-collection/S2-Level-2A-SR/collection.json and for a a 30 Day "quality" mosaic as shown in https://radiantearth.github.io/stac-browser/#/external/d29cp2gnktw6by.cloudfront.net/stac-collection/NDVI%20MAX%5BTEST%5D/collection.json the average computation is about 2000 and 5000 EECU seconds respectively for the month of Sept 2023. So, as long as we do not add more than 10-20% of additional EECU:

-> I vouch  for **Option 1** (map all images to the displacement function, everytime we generate a new product)

**Comment by Tschoun** [comment]
Did I understand that correctly, that option 1 would be different displacements for different products generated from the same image? Shouldn't we have a consistency between the products. I don't know what the consequences are, if storing or recalculating needs more computational power, but I think it depends on how we set up the processing pipeline. If we calculate each product by itself, then individual displacements make sense, but I don't see the advantage of running e.g. the cloud masking again when calculating an NDVI, when it was already done so for pure reflectance.

**Comment by davidoesch** [comment]
It is important that each product based on the same imagery has the same co-registration applied. And we also need the x/y shifts per raster cell in an additional layer as defined in the specification /pflichtenheft. The idea of the processing pipeline is that the set ofraw image is coregistered once and then  based on the coregistrated image the different products are derived. This can be achieved that registration matrix for the imagery is stored as additional layer.

When a rerun is done for one image or a time period (eg last 30day maxndvi) then the registration is redone. But only one per image. This is how I understood option 1

**Comment by Tschoun** [comment]
Definitely Option 2 then since we have to store the displacement vectors anyway.

**Comment by gilliM** [comment]
There would be no problem to store the displacement vector with each product in option 1.

The big difference between the options is that in option 2, we need to run the tasks in sequence. We need to be sure that the registration is done and that we have the asset available before proceeding.

That add a kind of step like this:
1 satromo_registration.py
2 satromo_processor.py
3 satromo_publisher.py

**Comment by davidoesch** [comment]
> Definitely Option 2 then since we have to store the displacement vectors anyway.

yep agree on that.

**Comment by gilliM** [comment]
Here is an important observation:

I did a short analysis of reproducibility of the displacement results. I used the sames images to produce different product sat different times, hopefully with  different context conditions on GEE servers.

It looks like the displacement results are fully reproduced when using the same raw images.
This would mean we could re-use independently the displacement function on the same images for different products.
The storage of displacement vectors could be attached to any product, for example the surface reflectance, and store independently at the end.

Let's discuss that during our next meeting.
